### PR TITLE
Document the line-number counting behavior

### PR DESCRIPTION
**Pull request by wxiaoguang**
See the comment. Ref #27377

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Review by CaiCandong — APPROVED** [review]

File: routers/web/repo/view.go [hunk 1]
```diff
@@ -488,8 +488,13 @@ func renderFile(ctx *context.Context, entry *git.TreeEntry, treeLink, rawLink st
 		} else {
 			buf, _ := io.ReadAll(rd)
 
-			// empty: 0 lines; "a": one line; "a\n": two lines; "a\nb": two lines;
-			// the NumLines is only used for the display on the UI: "xxx lines"
+			// The Open Group Base Specification: https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html
+			//   empty: 0 lines; "a": 1 line, 1 incomplete-line; "a\n": 1 line; "a\nb": 1 line, 1 incomplete-line;
```
Comment (delvh) [inline routers/web/repo/view.go:492]: The last one or the second one doesn't make sense.
How can they be equal even though there's a `\n` in between?

Comment (wxiaoguang) [reply routers/web/repo/view.go:492]: ~~I didn't get your point. What is the "sense" and what is "being equal"?~~

Comment (wxiaoguang) [reply routers/web/repo/view.go:492]: Hmm, I see, the comment is not right. Will propose a fix.

Comment (delvh) [reply routers/web/repo/view.go:492]: > "a": 1 line, 1 incomplete-line

> "a\nb": 1 line, 1 incomplete-line

Both have the same result, even though there is a `\n` in between?

Comment (wxiaoguang) [reply routers/web/repo/view.go:492]: ->  Show a no-EOL mark for the last line when viewing a file #27391

**Pull request merged**